cirq/ops/common_gates.py

```diff
@@ -379,6 +380,25 @@ def with_canonical_global_phase(self) -> 'ZPowGate':
         """Returns an equal-up-global-phase standardized form of the gate."""
         return ZPowGate(exponent=self._exponent)
 
+    def controlled(self,
+                   num_controls: int = None,
+                   control_values: Optional[Sequence[
+                       Union[int, Collection[int]]]] = None,
+                   control_qid_shape: Optional[Tuple[int, ...]] = None
+                  ) -> raw_types.Gate:
+        """
+        Specialize controlled for ZPow to return corresponding CZPow when
+        controlled by a single qubit.
+        """
+        if ((num_controls is None or num_controls == 1) and
```

Comment:
Avoid redundant boilerplate by delegating to `Gate.controlled` to fill in the values.

```
result = super().controlled(num_controls, control_values, control_qid_shape)
if result.control_qid_shape == (1,) and result.control_values == (0,):
    return cirq.CZPowGate(...)
return result
```